components/AvalancheForecastZoneMap.tsx

```diff
@@ -394,6 +400,26 @@ const AvalancheForecastZoneCards: React.FunctionComponent<{
 
   const flatListRef = useRef(null);
 
+  const handleScroll = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+    if (!offsets || offsets.length === 0) {
+      return;
+    }
+    // we want to figure out what card the user scrolled to - so, we can figure out which card
+    // offset the center of the screen is at; by moving through the list backwards we can simply
+    // exit when we find an offset that's before the center of the screen (we know since we got
+    // that far in the iteration that the offset prior to that is on the other side of the center
+    // of the screen)
+    let index = 0;
+    const offset = event.nativeEvent.contentOffset.x + width / 2;
+    for (let i = offsets.length - 1; i >= 0; i--) {
+      if (offsets[i] <= offset) {
+        index = i;
+        break;
+      }
+    }
```

Comment:
Does it work to just look for the first offset that's greater, and then go back one? (

```suggestion
    const cardCenter = event.nativeEvent.contentOffset.x + width / 2;
    const index = Math.max(0, offsets.findIndex(o => o > cardCenter) - 1);
```